strawberry/lazy_type.py

```diff
@@ -73,7 +73,7 @@ def __init__(self, module: str) -> None:
         self.package = None
 
         if module.startswith("."):
-            frame = inspect.stack()[2][0]
+            frame = inspect.currentframe().f_back.f_back
```

Comment:
I didn't bother to check for nulls at every location, since this should only be accessed via `.lazy()` and also the previous code would've hit a bounds check in the same condition anyways

Reply:
nice! we also use `sys._getframe(x)` in another place, would you mind trying with that and see if performs the same? 😊

Reply:
Yupp! It has the same effect - actually `_getframe` is in C, so looks like it is even faster. Whipped up a small benchmark to look at the difference:

<img width="723" alt="Screenshot 2023-07-06 at 5 34 35 PM" src="https://github.com/strawberry-graphql/strawberry/assets/2664912/80cb0715-93a4-41e3-8a65-9506309a9ad2">

<details>

	import time
	import inspect
	import sys

	def run_benchmark(name, fn):
		start_time = time.time()

		for i in range(1000000):
			fn()

		end_time = time.time()
		duration = end_time - start_time

		print(f"{name} completed in {duration}s")

	def main():
		run_benchmark("inspect.stack()[2]", lambda: inspect.stack()[2])
		run_benchmark("inspect.currentframe().f_back.f_back", lambda: inspect.currentframe().f_back.f_back)
		run_benchmark("sys._getframe(2)", lambda: sys._getframe(2))

	main()

</details>

Reply:
awesome! thanks for that!